src/library.js

```diff
@@ -1327,6 +1335,30 @@ const createPilingJs = (rootElement, initOptions = {}) => {
     }
 
     state = newState;
+
+    pubSub.publish('update');
+  };
+
+  const exportState = () => {
+    const clonedState = deepClone(state);
+    clonedState.version = VERSION;
+    return clonedState;
+  };
+
+  const importState = (importedState, overwriteState = false) => {
+    if (importedState.version !== VERSION) {
+      console.warn(
+        `The version of the imported state "${importedState.version}" doesn't match the library version "${VERSION}". Use at your own risk!`
+      );
+    }
+    delete importedState.version;
+    if (importedState.piles !== store.getState().piles)
+      store.dispatch(createAction.overwritePiles(importedState.piles));
```

Comment:
I don't quite understand why we need 2 separate actions here. If `overwriteState` is `true` then we anyway overwrite the entire state, including `piles`. If on the other hand `overwriteState` is `false` then we selectively loop over the new state and update the old state whenever something changed. So if the `piles` changed `newState.piles` should not equal `oldState.piles` is that not true?

The reason why I would not like to introduce multiple actions is that this can get hairy very quickly. What about other properties of the state? Do they need extra actions too? If our current `softOverwrite` can't handle `piles` correctly then we need to just edit this function.

Reply:
I see. I just didn't know how to edit `softOverwrite` function to correctly overwrite the state 😞Also, I still don't understand when should the 2 passes be done? After we overwrite the state? I'm still confused about the whole logic. Sorry for that 😥